modules/apps/dynamic-data-mapping/dynamic-data-mapping-expression/src/main/java/com/liferay/dynamic/data/mapping/expression/internal/DDMExpressionEvaluatorVisitor.java

```diff
@@ -136,6 +136,13 @@ public Object visitEqualsExpression(
 		Object object1 = visitChild(context, 0);
 		Object object2 = visitChild(context, 2);
 
+		if ((object1 instanceof Number) && (object2 instanceof Number)) {
+			BigDecimal bigDecimal1 = getBigDecimal(visitChild(context, 0));
+			BigDecimal bigDecimal2 = getBigDecimal(visitChild(context, 2));
```

Comment:
@mlan0, once you already have objects (`object1` and `object2`) holding the values, why not use them here instead of call the method `visitChild` again? Does is make sense to you?

Reply:
It does. I will change it!